tools/block-generator/runner/run.go

```diff
@@ -131,7 +174,20 @@ func (r *Args) run() error {
 	defer report.Close()
 
 	// Run the test, collecting results.
-	if err := r.runTest(report, indexerNet, algodNet); err != nil {
+	// check /metrics endpoint is available before running the test
+	var resp *http.Response
+	for retry := 0; retry < 10; retry++ {
+		resp, err = http.Get(fmt.Sprintf("http://%s/metrics", metricsNet))
```

Comment:
I noticed that '/metrics' isn't always available by the time runTest starts. this causes block-generator to exit with error `problem collecting metrics (1 / 3.003180017s): unable to read metrics url 'http://localhost:9999/metrics'`. 
I'm adding these lines to make sure `/metrics` is available before running tests.